src/common/s3_adapter.cpp

```diff
@@ -546,11 +552,10 @@ Aws::S3::Model::CompletedPart S3Adapter:: UploadOnePart(
     request.SetUploadId(uploadId);
     request.SetPartNumber(partNum);
     request.SetContentLength(partSize);
-    auto str = std::make_shared<std::string>(buf, partSize);
-    auto input_data =
-            Aws::MakeShared<Aws::StringStream>("UploadPartStream");
-    *input_data << *str;
-    request.SetBody(input_data);
+
+    request.SetBody(Aws::MakeShared<PreallocatedIOStream>(
+        AWS_ALLOCATE_TAG, const_cast<char *>(buf), partSize));
```

Comment:
Can we provide an PreallocatedIOStream constructor using param of const char *  buf to avoiding const_cast here?

Reply:
fix